subsys/bluetooth/host/gatt.c

```diff
@@ -2852,12 +2855,20 @@ struct bt_gatt_attr *bt_gatt_find_by_uuid(const struct bt_gatt_attr *attr,
 					  const struct bt_uuid *uuid)
 {
 	struct bt_gatt_attr *found = NULL;
-	uint16_t start_handle = bt_gatt_attr_value_handle(attr);
-	uint16_t end_handle = start_handle && attr_count ?
-			      start_handle + attr_count : 0xffff;
+	uint16_t start_handle = bt_gatt_attr_get_handle(attr);
+	uint16_t end_handle = start_handle && attr_count
+				      ? MIN(start_handle + attr_count, BT_ATT_LAST_ATTRIBUTE_HANDLE)
```

Comment:
This MIN-protection does not work since `start_handle + attr_count` will overflow first, right?

Reply:
Wasn't sure, and may be compiler/platform dependent, but checked this in a local test file (using GCC and x86-64), and it seemed to cast the result to an `int` which could store the overflowed result, before clamping it. 

```c
#include <stdint.h>
#include <stdio.h>

#define MIN(a, b) (((a) < (b)) ? (a) : (b))

int main() {
    uint8_t a = 200;
    uint8_t b = 100;
    uint8_t result;

    result = MIN(a + b, 234);
    printf("Result: %u\n", result);

    result = a + b;
    printf("Result: %u\n", result);
    return 0;
}
```

prints
```
Result: 234
Result: 44
```

Reply:
OK. It looks like C promotes integers smaller than `int` to an `int` when they are used with an operator. I.e. There is no `+` for `char`. This is true even for unary `+`.

Reply:
Yeah, so as long as A + B isn't larger than INT_MAX it should be OK

Reply:
`u16_add_overflow` could also have been used